pkg/apis/eksctl.io/v1alpha5/validation.go

```diff
@@ -111,6 +107,9 @@ func ValidateClusterConfig(cfg *ClusterConfig) error {
 
 // ValidateClusterEndpointConfig checks the endpoint configuration for potential issues
 func (c *ClusterConfig) ValidateClusterEndpointConfig() error {
+	if !c.HasClusterEndpointAccess() {
```

Comment:
@martina-if this check is fine, however the implementation of `HasClusterEndpointAccess` is a little bit weird for me

```
func (c *ClusterConfig) HasClusterEndpointAccess() bool {
	if c.VPC != nil && c.VPC.ClusterEndpoints != nil {
		pubAccess := c.VPC.ClusterEndpoints.PublicAccess
		privAccess := c.VPC.ClusterEndpoints.PrivateAccess
		hasPublicAccess := pubAccess != nil && *pubAccess
		hasPrivateAccess := privAccess != nil && *privAccess
		return hasPublicAccess || hasPrivateAccess
	}
	return true // so if nothing is specficified, it returns true, which is has cluster endpoint configured.
}
```